SteakBot.Core/Objects/MemeCommand.cs

```diff
@@ -19,5 +20,57 @@ public MemeCommand(MemeResultType resultType, string name, string value, string
             Value = value;
             Description = description;
         }
+
+        #region IEquitable
+
+        public override bool Equals(object other)
+        {
+            var item = other as MemeCommand;
+            return item != null && Equals(item);
+        }
+
+        public bool Equals(MemeCommand other)
+        {
+            if (ReferenceEquals(other, null))
+            {
+                return false;
+            }
+
+            return (object.ReferenceEquals(this.ResultType, other.ResultType)
+                    || this.ResultType.Equals(other.ResultType))
+                && (object.ReferenceEquals(this.Name, other.Name)
+                    || this.Name != null
+                    && this.Name.Equals(other.Name))
+                && (object.ReferenceEquals(this.Value, other.Value)
+                    || this.Value != null
+                    && this.Value.Equals(other.Value))
+                && (object.ReferenceEquals(this.Description, other.Description)
+                    || this.Description != null
+                    && this.Description.Equals(other.Description));
+        }
+
+        public override int GetHashCode()
+        {
+            return this.ResultType.GetHashCode()
```

Comment:
dont use XOR for combining hashes
Instance A: Prop1 = 1 Prop2 = 0 Instance B: Prop1 = 0 Prop2 = 1 => XOR produces equal hashses.